### ISSUE TITLE
Use of of "%" (percent) signs in bibliographies breaks citeproc

### ISSUE THREAD

**Issue by ucpresearch** [jgm/pandoc#8604]
**Problem** 

Pandoc 3.0.1 (problem does not happen in 2.19.2 or earlier versions) stops processing after percent signs in bibtex bibliographies. When processing files that contains percent signs leads to an error message on the following entry. 
```
(line ..., column 1):
unexpected '@'
```

**Expected behavior**

Simply processing the bib file as previous versions / bibtex would, with no errors. 

**How to reproduce**

I reproduced the example on https://pandoc.org/try/ using the code below. I downloaded it as a json file, which is attached to this bug report as a txt file (github limitations). This is a relatively minor bug for my use case.
[pandoc-percent-bug.txt](https://github.com/jgm/pandoc/files/10684271/pandoc-percent-bug.txt)

Thanks! 

Minimal bib file to reproduce:
```
@Misc{ent1,
   author = "Smith, Kim",
   year = 2003,
   abstract = {10% of the cases}
}

@Misc{ent2,
   author = "Smith, Sandy",
   year = 2002,
   abstract = {another entry}
}
```

Minimal md file:
```
This is true [@ent1]
```

**Comment by tarleb** [comment]
Duplicate of #8595. This is already fixed in the development version.

**Comment by tarleb** [comment]
Correction: this is **not** a duplicate of #8595, but it's not a bug either. The use of an unescaped percent sign in this field is not allowed by BibLaTeX, so it seems ok that pandoc fails here as well.

**Comment by ucpresearch** [comment]
But bibtex doesn't mind it. Is the goal to be compliant with biblatex and not bibtex?

**Comment by ucpresearch** [comment]
Previous versions of pandoc didn't mind it either. My own usage case is solved already (I replaced percent signs with the word percent), so I have no personal stake in the matter. 

Either way, I am thankful for pandoc and its development. Thanks for responding quickly too.

**Comment by tarleb** [comment]
> But bibtex doesn't mind it. Is the goal to be compliant with biblatex and not bibtex?

Oh, I didn't check for that. Let me reopen this until we are sure that pandoc's behavior is as intended.

**Comment by jgm** [comment]
Exactly this issue was just fixed earlier this week - #8597 - remember to check closed issues!

**Comment by jgm** [comment]
By the way, if your bibliography style actually *uses* the abstract field, this bibtex file *will* cause problems (with both latex and pandoc).  So I'd recommend escaping the `%`.

**Comment by ucpresearch** [comment]
> Exactly this issue was just fixed earlier this week - #8597.

My apologies! I looked for this bug, but I must have looked only among the open ones. 

> By the way, if your bibliography style actually uses the abstract field, this bibtex file will cause problems (with both latex and pandoc). So I'd recommend escaping the `%`.

Thanks for the warnings. Many publishers let you download the abstract with the reference, and I found it useful to keep them in. I don't use them except as a database / auctex lookup. 

I already thanked @tarleb above, but of course this is meant to you too. I love pandoc, thanks for all the hard work.
